source/Applications/Advanced/MultiCamera/MultiCameraCaptureInParallel/MultiCameraCaptureInParallel.cpp

```diff
@@ -0,0 +1,76 @@
+/*
+This example shows how to capture point clouds with multiple cameras in parallel.
+*/
+
+#include <Zivid/Zivid.h>
+
+#include <chrono>
+#include <iostream>
+#include <mutex>
+#include <thread>
+#include <vector>
+
+namespace
+{
+    void captureInThread(Zivid::Camera &camera, std::mutex &m)
+    {
+        const auto parameters = Zivid::CaptureAssistant::SuggestSettingsParameters{
+            Zivid::CaptureAssistant::SuggestSettingsParameters::AmbientLightFrequency::none,
+            Zivid::CaptureAssistant::SuggestSettingsParameters::MaxCaptureTime{ std::chrono::milliseconds{ 800 } }
+        };
+        const auto settings = Zivid::CaptureAssistant::suggestSettings(camera, parameters);
+
+        // Capturing frame
+        auto frame = camera.capture(settings);
+
+        const auto dataFile = "Frame_" + camera.info().serialNumber().value() + ".zdf";
+
+        // Saving frame to file
+        std::unique_lock<std::mutex> lock(m);
```

Comment:
Where is the unlock?

Reply:
Ahh.. this is for single execution, and you're just protecting file write? Question 1: Is it required to protect file write? Question 2: Should we still explicitly unlock the mutex?

Reply:
I see that the destructor of std:unique_lock unlocks the mutex, so that's quite explicit.

Reply:
1) Yeah writing to file failed every time I didn't do mutex. 
2) yes it unlocks as you say